src/pages/instances/forms/CreateInstanceFromSnapshotForm.tsx

```diff
@@ -0,0 +1,308 @@
+        {instanceName}
+      </Link>
+    );
+
+    const message = <>Created Instance {instanceLink}.</>;
+    toastNotify.success(message);
+  };
+
+  const getDuplicatedInstanceName = (instance: LxdInstance): string => {
+    const instanceNames = instances.map((instance) => instance.name);
+    const newInstanceName = truncateInstanceName(
+      instance.name,
+      `-${snapshot.name}-copy`,
+    );
+
+    if (instanceNames.includes(newInstanceName)) {
+      let count = 1;
+      while (instanceNames.includes(`${newInstanceName}-${count}`)) {
+        count++;
+      }
+      return `${newInstanceName}-${count}`;
+    }
+
+    return newInstanceName;
+  };
+
+  const validate = (values: CreateInstanceFromSnapshotValues) => {
+    const errors: Partial<CreateInstanceFromSnapshotValues> = {};
+    if (values.stateful) {
+      if (values.targetProject === instance.project) {
+        errors.targetProject =
+          "Stateful instance snapshot duplication must be in a different project";
+      }
+
+      if (values.instanceName !== instance.name) {
+        errors.instanceName =
+          "Instance name must be the same for stateful duplication";
+      }
+    }
+
+    return errors;
+  };
+
+  const formik = useFormik<CreateInstanceFromSnapshotValues>({
+    initialValues: {
+      instanceName: getDuplicatedInstanceName(instance),
+      targetProject: instance.project,
+      stateful: false,
+      targetClusterMember: isClustered ? instance.location : "",
+      targetStoragePool:
+        (instance.devices["root"] as LxdDiskDevice)?.pool ??
+        storagePools[0]?.name,
+    },
+    enableReinitialize: true,
+    validate,
+    validationSchema: Yup.object().shape({
+      instanceName: instanceNameValidation(
+        instance.project,
+        controllerState,
+        instance.name,
+      ).required(),
+    }),
+    onSubmit: (values) => {
+      createInstance(
+        JSON.stringify(instanceFromSnapshotPayload(values, instance, snapshot)),
+        values.targetProject,
+        values.targetClusterMember,
+      )
+        .then((operation) => {
+          toastNotify.info(
+            `Instance creation started for ${values.instanceName}.`,
+          );
+          eventQueue.set(
+            operation.metadata.id,
+            () => notifySuccess(values.instanceName, values.targetProject),
+            (msg) =>
+              toastNotify.failure(
+                "Instance creation failed.",
+                new Error(msg),
+                <InstanceLink instance={instance} />,
+              ),
+          );
+        })
+        .catch((e) => {
+          toastNotify.failure(
+            "Instance creation failed.",
+            e,
+            <InstanceLink instance={instance} />,
+          );
+        })
+        .finally(() => {
+          close();
+        });
+    },
+  });
+
+  return (
+    <Modal
+      close={close}
+      className="duplicate-instances-modal"
```

Comment:
```suggestion
```
Seems unused.

Reply:
I reused the class for styling, I've added an extra class `create-instance-from-snapshot-modal` now to avoid any confusion.